windows/src/developer/TIKE/dialogs/UfrmVisualKeyboardImportKMX.pas

```diff
@@ -240,16 +254,37 @@ procedure TfrmVisualKeyboardImportKMX.SendKey;
 //    else lb.items.Add(Format('SendKey: nkey: %d vkey: %s shift: %d ansi', [nkey, VKeyNames[vk.vkey], vk.kmshift]));
 //  ShowMessage(Format('Sending key: %s / %d', [VKeyNames[vk.VKey], vk.shift]));
 
+  n := GetVKLegalShiftStateIndex(vk.shift);
+  if n < 0
+    then shiftText := Format('%x', [vk.shift])
+    else shiftText := VKLegalShiftStates[n].Desc;
+
+  if vk.vkey < 256
+    then keyText := VKeyNames[vk.vkey]
+    else keyText := Format('%x', [vk.vkey]);
+
+  SetStatus(Format('Importing %s %s', [shiftText, keyText]));
+
   n := 0;
 
   if (vk.kmshift and KMX_SHIFTFLAG) = KMX_SHIFTFLAG then AddInput(VK_SHIFT, True, False);   // I4143
   if (vk.kmshift and KMX_RALTFLAG) = KMX_RALTFLAG then AddInput(VK_MENU, True, True);
+  if (vk.kmshift and KMX_LALTFLAG) = KMX_LALTFLAG then AddInput(VK_MENU, True, False);   // I4156
+  if (vk.kmshift and KMX_RCTRLFLAG) = KMX_RCTRLFLAG then AddInput(VK_CONTROL, True, True);
+  if (vk.kmshift and KMX_LCTRLFLAG) = KMX_LCTRLFLAG then AddInput(VK_CONTROL, True, False);   // I4156
+
   if (vk.kmshift and KMX_CTRLFLAG) = KMX_CTRLFLAG then AddInput(VK_CONTROL, True, False);
   if (vk.kmshift and KMX_ALTFLAG) = KMX_ALTFLAG then AddInput(VK_MENU, True, False);   // I4156
+
   AddInput(vk.vkey, True, False);
   AddInput(vk.vkey, False, False);
+
   if (vk.kmshift and KMX_ALTFLAG) = KMX_ALTFLAG then AddInput(VK_MENU, False, False);   // I4156
   if (vk.kmshift and KMX_CTRLFLAG) = KMX_CTRLFLAG then AddInput(VK_CONTROL, False, False);
+
+  if (vk.kmshift and KMX_LCTRLFLAG) = KMX_LCTRLFLAG then AddInput(VK_CONTROL, False, False);   // I4156
+  if (vk.kmshift and KMX_RCTRLFLAG) = KMX_RCTRLFLAG then AddInput(VK_CONTROL, False, True);
+  if (vk.kmshift and KMX_LALTFLAG) = KMX_LALTFLAG then AddInput(VK_MENU, False, False);   // I4156
   if (vk.kmshift and KMX_RALTFLAG) = KMX_RALTFLAG then AddInput(VK_MENU, False, True);
   if (vk.kmshift and KMX_SHIFTFLAG) = KMX_SHIFTFLAG then AddInput(VK_SHIFT, False, False);
 
```

Comment:
In case somebody else needs to maintain this block (lines 270-287), is there a specific order of operations they need to do these AddInputs?  I see it's currently LIFO.